pwnlib/libcdb.py

```diff
@@ -126,7 +127,23 @@ def provider_local_system(hex_encoded_id, hash_type):
         return local_libc.data
     return None
 
-PROVIDERS = [provider_local_system, provider_libcdb, provider_libc_rip]
+# Offline search https://github.com/niklasb/libc-database for hash type
+def provider_local_database(hex_encoded_id, hash_type):
+    assert context.local_libcdb
+
+    localdb = Path(context.local_libcdb)
+    if not localdb.is_dir():
+        log.warn_once("%s does not exist, please download libc-database first.", str(localdb))
+        return None
+
+    log.debug("Searching local libc database, %s: %s", hash_type, hex_encoded_id)
+    for libc_path in localdb.rglob("*.so"):
+        if hex_encoded_id == HASHES[hash_type](libc_path):
+            return read(libc_path)
+
+    return None
+
+PROVIDERS = [provider_local_database, provider_local_system, provider_libcdb, provider_libc_rip]
```

Comment:
Testing the local system's libc first seems faster than running through the libc database
```suggestion
PROVIDERS = [provider_local_system, provider_local_database, provider_libcdb, provider_libc_rip]
```